### PR TITLE
Add StringPrefix filter with tests

### PR DESCRIPTION
**Pull request by thexpand**
https://github.com/zendframework/zend-filter/issues/28

### CODE REVIEW

File: test/StringPrefixTest.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+     * @return void
+     */
+    public function testBasic()
+    {
+        $filter = $this->filter;
+
+        $prefix = 'ABC123';
+        $filter->setPrefix($prefix);
+
+        $this->assertStringStartsWith($prefix, $filter('sample'));
```
Comment (waltertamboer) [inline test/StringPrefixTest.php]: Assert methods are static so calling them with `$this->` is not correct.

Comment (thexpand) [reply test/StringPrefixTest.php]: @waltertamboer I'll fix this next week, because I'm on holiday.

Comment (froschdesign) [reply test/StringPrefixTest.php]: @waltertamboer @thexpand 
No change is needed here. `$this->` is basically used everywhere in tests.

Comment (froschdesign) [reply test/StringPrefixTest.php]: Background: https://discourse.zendframework.com/t/this-assert-vs-self-assert/448

Comment (waltertamboer) [reply test/StringPrefixTest.php]: Read it. I still don't get why we would promote bad syntax. It's not the standard I would expect from Zend Framework.

Comment (froschdesign) [reply test/StringPrefixTest.php]: > It's not the standard I would expect from Zend Framework.

"There is no right way": https://phpunit.readthedocs.io/en/7.3/assertions.html#static-vs-non-static-usage-of-assertion-methods

Comment (waltertamboer) [reply test/StringPrefixTest.php]: I don't agree. There is a right way. The paragraphs in the phpunit docs are just extremely awkward to read.

> The short answer is: there is no right way. And there is no wrong way, either. It is a matter of personal preference.

Which can be rephrased as: "it doesn't matter". No, it's not a matter of personal preference. It's a matter of writing semantically correct code. Basically the arguments @Ocramius used in the discussion on Discourse.

But it's ok, I've made my argument and it won't change the fact that ZF is currently using `$this`. So let's move on.

Thanks for your time and patience.

Comment (froschdesign) [reply test/StringPrefixTest.php]: @waltertamboer 
I promote nothing here, it is a purely pragmatic decision: It is a code review and not a coding standard or semantically correct code discussion.

Please do not get me wrong, your comments are important, but this is the wrong place and after the merge of this pull request your comments are lost.

(_end of offtopic_)

File: test/StringSuffixTest.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public function testWithoutSuffix()
+    {
+        $filter = $this->filter;
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('expects a "suffix" option; none given');
+        $filter('sample');
+    }
+
+    public function testNonStringSuffix()
```
[the remark this answers is not in the document]
Comment (thexpand) [reply test/StringSuffixTest.php]: I have added all possible PHP types (excluding string) in a data-provider.

File: test/StringPrefixTest.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    public function testWithoutPrefix()
+    {
+        $filter = $this->filter;
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('expects a "prefix" option; none given');
+        $filter('sample');
+    }
+
+    public function testNonStringPrefix()
```
[the remark this answers is not in the document]
Comment (thexpand) [reply test/StringPrefixTest.php]: I have added all possible PHP types (excluding string) in a data-provider.

**Comment by weierophinney** [comment]
Thanks, @thexpand! I have added documentation and merged for release!

**Pull request merged**